### PR TITLE
Add sa-SDR loss function

### PR DESCRIPTION
**Pull request by thequilo**

### CODE REVIEW

File: padertorch/ops/losses/regression.py [hunk 1]
```diff
@@ -277,5 +277,21 @@ def log1p_mse_loss(estimate: torch.Tensor, target: torch.Tensor,
...
 
-# TODO: Add log1p_mse loss from interspeech paper
+
+def source_aggregated_sdr_loss(
+        estimate: torch.Tensor,
+        target: torch.Tensor,
+) -> torch.Tensor:
+    """
+    The source-aggregated SDR loss. There is no `reduction` argument because the loss always
+    combines the losses for multiple paris of targets and estimates.
```
Comment (boeddeker) [inline padertorch/ops/losses/regression.py:287]: `paris` -> `pairs`

File: padertorch/ops/losses/regression.py [hunk 2]
```diff
@@ -277,5 +277,21 @@ def log1p_mse_loss(estimate: torch.Tensor, target: torch.Tensor,
...
     )
 
-# TODO: Add log1p_mse loss from interspeech paper
+
+def source_aggregated_sdr_loss(
+        estimate: torch.Tensor,
+        target: torch.Tensor,
+) -> torch.Tensor:
+    """
+    The source-aggregated SDR loss. There is no `reduction` argument because the loss always
```
Comment (boeddeker) [inline padertorch/ops/losses/regression.py:286]: Maybe you can highlight, that the loss is equal for reduction `mean` and `sum`?

File: padertorch/ops/losses/regression.py [hunk 3]
```diff
@@ -277,5 +277,21 @@ def log1p_mse_loss(estimate: torch.Tensor, target: torch.Tensor,
...
+    combines the losses for multiple paris of targets and estimates.
+    """
+    # Calculate the source-aggregated SDR: Sum the squares of all targets and
+    # all errors before computing the ratio.
+    sa_sdr = 10 * torch.log10(
+        torch.sum(target**2) / torch.sum((estimate - target)**2)
```
Comment (boeddeker) [inline padertorch/ops/losses/regression.py:292]: Question: Is mean more stable?

Comment (thequilo) [reply padertorch/ops/losses/regression.py:292]: I don't think that it makes a difference. I just tested with very large input shapes and input values and couldn't observe a difference larger than 1e-6.

Comment (boeddeker) [reply padertorch/ops/losses/regression.py:293]: ok, great. I am not sure, if an overflow may be a problem for float32, but I also don't know if the mean is overflow stable.

**Review by boeddeker — APPROVED** [review]
